### PR TITLE
Update BBEs with streaming usages

### PR DESCRIPTION
**Pull request by chamil321**

### CODE REVIEW

File: examples/response-with-multiparts/response_with_multiparts.bal [hunk 1]
```diff
@@ -119,7 +119,22 @@ function handleContent(mime:Entity bodyPart) {
...
             log:printError("Error in parsing text data", err = payload);
         }
-    }
+    } else if (mime:APPLICATION_PDF == baseType) {
+        // [Extracts byte stream](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/http/latest/http/classes/Response#getByteStream) from the body part and save it as a file.
+        var payload = bodyPart.getByteStream();
+        if (payload is stream<byte[], io:Error>) {
+            //Writes the incoming stream to a file using `io:fileWriteBlocksFromStream` API by providing the file location which the content should be written to.
+            io:Error? result = io:fileWriteBlocksFromStream(
+                                    "./files/ReceivedFile.pdf", streamer);
```
Comment (ayeshLK) [inline examples/response-with-multiparts/response_with_multiparts.bal:128]: It will be more readable if we take this to one line.

File: examples/http-streaming/http_streaming.bal [hunk 2]
```diff
@@ -40,27 +35,25 @@ service /'stream on new http:Listener(9090) {
     }
 
     resource function post receiver(http:Caller caller,
-                                        http:Request clientRequest) {
+                                        http:Request request) {
```
[the remark this answers is not in the document]
Comment (chamil321) [reply examples/http-streaming/http_streaming.bal:38]: These bal files are used to generate BBE site examples. So there is a maximum column with for BBE ball that is 80. If is exceeds the HTML generation fails. That is the reason not to have args on the same line.

File: examples/response-with-multiparts/response_with_multiparts.bal (hunk 1 above)
Comment (chamil321) [reply examples/response-with-multiparts/response_with_multiparts.bal:128]: Same reason applicable here as well

File: examples/different-payload-types/different_payload_types.bal [hunk 3]
```diff
@@ -47,23 +47,39 @@ service /actionService on new http:Listener(9090) {
...
-        //with a body parts as the payload.
-        mime:Entity[] bodyParts = [part1, part2];
-        response = clientEP->post("/echo", bodyParts);
-        handleResponse(response);
+        //Get a byte stream to a given file.
+        var bStream = io:fileReadBlocksAsStream("./files/logo.png");
+
+        if (bStream is stream<byte[], io:Error>) {
+            //POST remote function with byte stream as payload. Since the file path is static
+            //`untaint` is used to denote that the byte stream is trusted .
```
Comment (pubudu91) [inline examples/different-payload-types/different_payload_types.bal:55]: remote function -> remote method
`untaint` -> `<@untainted>`
Extra space at the end: "trusted ."

Comment (chamil321) [inline examples/different-payload-types/different_payload_types.bal:55]:
```suggestion
            //`<@untainted>` is used to denote that the byte stream is trusted.
```

File: examples/different-payload-types/different_payload_types.bal [hunk 4]
```diff
@@ -47,23 +47,39 @@ service /actionService on new http:Listener(9090) {
...
-        //[POST](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/http/latest/http/clients/Client#post) remote function
-        //with a body parts as the payload.
-        mime:Entity[] bodyParts = [part1, part2];
-        response = clientEP->post("/echo", bodyParts);
-        handleResponse(response);
+        //Get a byte stream to a given file.
+        var bStream = io:fileReadBlocksAsStream("./files/logo.png");
+
+        if (bStream is stream<byte[], io:Error>) {
+            //POST remote function with byte stream as payload. Since the file path is static
```
Comment (chamil321) [inline examples/different-payload-types/different_payload_types.bal:54]:
```suggestion
            //POST remote method with byte stream as payload. Since the file path is static
```

File: examples/different-payload-types/different_payload_types.bal (hunk 3 above)
Comment (pubudu91) [reply examples/different-payload-types/different_payload_types.bal:55]: Also, maybe rephrase this as either "Make a POST request with a byte stream as the payload" or "Call the `post()` method with a byte stream payload"

File: examples/different-payload-types/different_payload_types.bal [hunk 5]
```diff
@@ -47,23 +47,39 @@ service /actionService on new http:Listener(9090) {
...
+            //[Create a JSON body part](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/mime/latest/mime/classes/Entity#setJson).
+            mime:Entity part1 = new;
+            part1.setJson({"name": "Jane"});
+
+            //[Create a text body part](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/mime/latest/mime/classes/Entity#setText).
+            mime:Entity part2 = new;
+            part2.setText("Hello");
+
+            //[POST](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/http/latest/http/clients/Client#post) remote function
+            //with a body parts as the payload.
```
Comment (pubudu91) [inline examples/different-payload-types/different_payload_types.bal:68]: Similar comment as above regarding how this is phrased. And should be "with body parts"

File: examples/different-payload-types/different_payload_types.bal [hunk 6]
```diff
@@ -47,23 +47,39 @@ service /actionService on new http:Listener(9090) {
...
+            mime:Entity[] bodyParts = [part1, part2];
+            response = clientEP->post("/echo", bodyParts);
+            handleResponse(response);
 
-        var result = caller->respond(
-                                "Client actions successfully executed!");
-        handleError(result);
+            var result = caller->respond(
+                                    "Client actions successfully executed!");
```
Comment (pubudu91) [inline examples/different-payload-types/different_payload_types.bal:74]: This is just one action right?

File: examples/different-payload-types/different_payload_types.bal (hunk 4 above)
Comment (chamil321) [inline examples/different-payload-types/different_payload_types.bal:54]:
```suggestion
            //Make a POST request with a byte stream as the payload. Since the file path is static
```

File: examples/http-streaming/http_streaming.bal [hunk 7]
```diff
@@ -40,27 +35,25 @@ service /'stream on new http:Listener(9090) {
...
         http:Response res = new;
-        var payload = clientRequest.getByteChannel();
-        if (payload is io:ReadableByteChannel) {
-            //Writes the incoming stream to a file. First, [get the destination channel](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/io/latest/io/functions#openWritableFile)
-            //by providing the file name to which the content should be written to.
-            var destinationChannel =
-                io:openWritableFile("./files/ReceivedFile.pdf");
+        //[Retrieve the byteStream](https://ballerina.io/swan-lake/learn/api-docs/ballerina/#/ballerina/http/latest/http/classes/Request#getByteStream).
```
Comment (pubudu91) [inline examples/http-streaming/http_streaming.bal:40]: byteStream -> byte stream ?

File: examples/http-streaming/http_streaming.bal [hunk 8]
```diff
@@ -40,27 +35,25 @@ service /'stream on new http:Listener(9090) {
...
+        stream<byte[], io:Error>|error streamer = request.getByteStream();
+
+        if (streamer is stream<byte[], io:Error>) {
+            //Writes the incoming stream to a file using `io:fileWriteBlocksFromStream` API by providing the file location which the content should be written to.
```
Comment (pubudu91) [inline examples/http-streaming/http_streaming.bal:44]: "file location to which the content"

File: examples/http-streaming/http_streaming.description [hunk 9]
```diff
@@ -1,3 +1,3 @@
-//Ballerina supports HTTP input and output streaming capability through io:ReadableByteChannel.<br/><br/>
+//Ballerina supports HTTP input and output streaming capability through ballerina stream.<br/><br/>
```
Comment (pubudu91) [inline examples/http-streaming/http_streaming.description:1]: I think we may need to rephrase this. @MaryamZi thoughts?

File: examples/response-with-multiparts/response_with_multiparts.bal [hunk 10]
```diff
@@ -119,6 +119,22 @@ function handleContent(mime:Entity bodyPart) {
         } else {
             log:printError("Error in parsing text data", err = payload);
         }
+    } else if (mime:APPLICATION_PDF == baseType) {
+        // [Extracts byte stream](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/http/latest/http/classes/Response#getByteStream) from the body part and save it as a file.
```
Comment (pubudu91) [inline examples/response-with-multiparts/response_with_multiparts.bal:123]: save -> saves

File: examples/response-with-multiparts/response_with_multiparts.bal [hunk 11]
```diff
@@ -119,6 +119,22 @@ function handleContent(mime:Entity bodyPart) {
...
+        var payload = bodyPart.getByteStream();
+        if (payload is stream<byte[], io:Error>) {
+            //Writes the incoming stream to a file using `io:fileWriteBlocksFromStream` API by providing the file location which the content should be written to.
```
Comment (pubudu91) [inline examples/response-with-multiparts/response_with_multiparts.bal:126]: "location to which the content"

File: examples/response-with-multiparts/response_with_multiparts.bal [hunk 12]
```diff
@@ -119,6 +119,22 @@ function handleContent(mime:Entity bodyPart) {
...
+            io:Error? result = io:fileWriteBlocksFromStream(
+                                    "./files/ReceivedFile.pdf", payload);
+
+            if (result is error) {
+                log:printError("Error occurred while writing ",
```
Comment (pubudu91) [inline examples/response-with-multiparts/response_with_multiparts.bal:131]: Shall we use consistent cases for these log messages? Above, these log messages started with a lower case 'e'

**Review by Bhashinee — APPROVED** [review]

File: examples/different-payload-types/different_payload_types.bal [hunk 13]
```diff
@@ -47,23 +47,38 @@ service /actionService on new http:Listener(9090) {
...
-        //with a body parts as the payload.
-        mime:Entity[] bodyParts = [part1, part2];
-        response = clientEP->post("/echo", bodyParts);
-        handleResponse(response);
+        //Get a byte stream to a given file.
+        var bStream = io:fileReadBlocksAsStream("./files/logo.png");
+
+        if (bStream is stream<byte[], io:Error>) {
+            //Make a POST request with a byte stream as the payload. Since the file path is static
+            //`<@untainted>` is used to denote that the byte stream is trusted.
```
Comment (MaryamZi) [inline examples/different-payload-types/different_payload_types.bal:55]:
```suggestion
            // Make a POST request with a byte stream as the payload. Since the file path is static
            // `<@untainted>` is used to denote that the byte stream is trusted.
```

Shall we fix everywhere?

File: examples/different-payload-types/different_payload_types.bal [hunk 14]
```diff
@@ -47,23 +47,38 @@ service /actionService on new http:Listener(9090) {
...
+
+            //[Create a JSON body part](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/mime/latest/mime/classes/Entity#setJson).
+            mime:Entity part1 = new;
+            part1.setJson({"name": "Jane"});
+
+            //[Create a text body part](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/mime/latest/mime/classes/Entity#setText).
+            mime:Entity part2 = new;
+            part2.setText("Hello");
+
+            //Make a POST request with a body parts as the payload.
```
Comment (pubudu91) [inline examples/different-payload-types/different_payload_types.bal:67]:
```suggestion
            //Make a POST request with body parts as the payload.
```

File: examples/http-streaming/http_streaming.bal [hunk 15]
```diff
@@ -4,7 +4,7 @@ import ballerina/log;
 import ballerina/mime;
 
 // Creates an endpoint for the [client](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/http/latest/http/clients/Client).
-http:Client clientEndpoint = new ("http://localhost:9090");
+http:Client clientEndpoint = check new("http://localhost:9090");
```
Comment (MaryamZi) [inline examples/http-streaming/http_streaming.bal:7]:
```suggestion
http:Client clientEndpoint = check new ("http://localhost:9090");
```

File: examples/http-streaming/http_streaming.description [hunk 16]
```diff
@@ -1,3 +1,3 @@
-//Ballerina supports HTTP input and output streaming capability through io:ReadableByteChannel.<br/><br/>
+//Ballerina supports HTTP input and output streaming capability via ballerina stream functionality.<br/><br/>
```
Comment (MaryamZi) [inline examples/http-streaming/http_streaming.description:1]: How about something like
```suggestion
// Ballerina supports HTTP input and output streaming capability based on the Ballerina `stream` type.<br/><br/>
```

File: examples/http-streaming/http_streaming.bal [hunk 17]
```diff
@@ -76,42 +69,10 @@ function setError(http:Response res, error err) {
...
-    var cr = channelResult.close();
+//Closes the byte stream.
+function close(stream<byte[], io:Error> byteStream) {
+    var cr = byteStream.close();
     if (cr is error) {
-        log:printError("Error occurred while closing the channel: ", err = cr);
+        log:printError("Error occurred while closing the stream: ", err = cr);
     }
-}
+}
```
Comment (MaryamZi) [inline examples/http-streaming/http_streaming.bal:78]: Missing a new line.

File: examples/response-with-multiparts/response_with_multiparts.bal (hunk 12 above)
Comment (chamil321) [reply examples/response-with-multiparts/response_with_multiparts.bal:131]: Throughout this BBE, all error messages have started with a capital letter. Hence thought of keeping it as it is.
